[IMP] http_routing: remove trailing / for homepage url prefixed with …

rdeodoo opened the pull request:
…language

robodoo commented:
[Pull request status dashboard](https://mergebot.odoo.com/odoo/odoo/pull/71065).

rdeodoo commented:
To do: add tests redirect, something like
/fr_BE/ -> /fr_BE
/fr_BE/?a=b -> /fr_BE?a=b
CHeck if controller redirect are already tests
/my/ -> /my
/my/?a=b -> /my?a=b

rdeodoo commented on addons/http_routing/models/ir_http.py:
```diff
@@ -488,6 +488,11 @@ def _dispatch(cls):
                 elif url_lg:
                     request.uid = None
+                    if request.httprequest.path == '/%s/' % url_lg:
+                        # special case for homepage controller
```
more detailed comment + comment it is same as post_process_args + missing if querystring before append

rdeodoo commented:
- Tests added
- Found and fixed a similar bug for website.page _serve_page trailing slashes
- Removed trailing slash in controller only for frontend routes (removed the diff from backend routes)
- Removed the logger warning when a route had a trailing slash

seb-odoo commented on addons/web/controllers/main.py:
```diff
@@ -1423,7 +1423,7 @@ def content_assets(self, id=None, filename=None, unique=None, extra=None, **kw):
         '/web/partner_image/<int:rec_id>',
         '/web/partner_image/<int:rec_id>/<string:field>',
-        '/web/partner_image/<int:rec_id>/<string:field>/<string:model>/'], type='http', auth="public")
+        '/web/partner_image/<int:rec_id>/<string:field>/<string:model>'], type='http', auth="public")
```
Hello, just a heads up this route should be removed in https://github.com/odoo/odoo/pull/69819 which is going to be merged soon

JKE-be commented:
@robodoo r+ rebase-ff

robodoo commented:
Merge method set to rebase and fast-forward
